crates/db_connection_pool/src/postgrespool.rs

```diff
@@ -93,14 +109,32 @@ impl PostgresConnectionPool {
                     connection_string.push_str(format!("port={pg_port} ").as_str());
                 }
                 if let Some(pg_sslmode) = params.get("pg_sslmode") {
-                    connection_string.push_str(format!("sslmode={pg_sslmode} ").as_str());
+                    match pg_sslmode.to_lowercase().as_str() {
+                        "disable" | "require" | "prefer" => {
+                            ssl_mode = pg_sslmode.as_str();
+                        }
+                        _ => {
+                            InvalidParameterSnafu {
+                                parameter_name: "pg_sslmode".to_string(),
+                            }
+                            .fail()?;
+                        }
+                    }
                 }
-                if let Some(pg_insecure) = params.get("pg_insecure") {
-                    accept_invalid_certs = pg_insecure == "true";
+                if let Some(pg_sslrootcert) = params.get("pg_sslrootcert") {
+                    if !std::path::Path::new(pg_sslrootcert).exists() {
```

Comment:
This can be simplified to:

```rust
ensure!(std::path::Path::new(pg_sslrootcert).exists(), InvalidRootCertPathSnafu {
                            path: pg_sslrootcert,
                        });`
```

Reply:
thanks, updated!